de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetBetaWarning.java

```diff
@@ -0,0 +1,251 @@
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.core.runtime.QualifiedName;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.edit.domain.EditingDomain;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.jface.viewers.ISelectionChangedListener;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.ui.forms.widgets.FormToolkit;
+
+import de.dlr.sc.virsat.external.lib.commons.cli.Activator;
+import de.dlr.sc.virsat.model.dvlm.categories.CategoryAssignment;
+import de.dlr.sc.virsat.model.dvlm.concepts.Concept;
+import de.dlr.sc.virsat.model.dvlm.structural.StructuralElementInstance;
+import de.dlr.sc.virsat.uiengine.ui.editor.snippets.AUiSectionSnippet;
+import de.dlr.sc.virsat.uiengine.ui.editor.snippets.IUiSnippet;
+
+/**
+ *
+ * A snippet that warns if the current element is a BETA element or contains one
+ *
+ */
+public class UiSnippetBetaWarning extends AUiSectionSnippet implements IUiSnippet {
+
+	private static final String SECTION_HEADING = "Beta Warning!";
+	private static final int UI_LAYOUT_NR_COLUMNS = 1;
+	private static final String IGNORE_WARNING = "ignoreBetaConcepts";
+
+	private final boolean ignoreWarning;
+
+	/**
+	 * Constructor for the beta warning snippet
+	 */
+	public UiSnippetBetaWarning() {
+		super();
+		ignoreWarning = Activator.getCommandLineManager().isCommandLineOptionSet(IGNORE_WARNING);
+	}
+
+	@Override
+	public void createSwt(FormToolkit toolkit, EditingDomain editingDomain, Composite composite, EObject initModel) {
+		super.createSwt(toolkit, editingDomain, composite, initModel);
+
+		Composite sectionBody = createSectionBody(toolkit, SECTION_HEADING, null, UI_LAYOUT_NR_COLUMNS);
+		Label label = toolkit.createLabel(sectionBody, createDescriptionLabel());
+
+		// Configure color of the area to be red
+		Display display = Display.getCurrent();
+		Color red = display.getSystemColor(SWT.COLOR_DARK_RED);
+		Color gray = display.getSystemColor(SWT.COLOR_GRAY);
+		sectionBody.setBackground(red);
+		label.setBackground(red);
+		label.setForeground(gray);
+
+		setUpLabel(label);
+
+	}
+
+	/**
+	 * Method to set up the label
+	 * 
+	 * @param label
+	 *            The label to be set up
+	 */
+	private void setUpLabel(Label label) {
+		GridData gridData = createDefaultGridData();
+		gridData.horizontalSpan = 1;
+		label.setLayoutData(gridData);
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see
+	 * de.dlr.sc.virsat.uiengine.ui.editor.snippets.IUiSnippet#isActive(org.eclipse.
+	 * emf.ecore.EObject)
+	 */
+	@Override
+	public boolean isActive(EObject model) {
+
+		// If the command line option to ignore the warning is set then never show this
+		// snippet
+		if (ignoreWarning) {
+			return false;
+		}
+
+		if (model instanceof StructuralElementInstance) {

```

Comment:
Let's maybe simplify it like
`return !ignoreWarning && !getRelevantBetaConcept().isEmpty();`